controllers/raffleController.js

```diff
@@ -91,8 +96,27 @@ raffleController.post(
   async (request, response) => {
     const r = request.body;
     try {
-      const newRow = await addNewParticipant(r, request.params.id);
-      if (newRow) response.status(201).json({ data: newRow });
+      const result = await getRaffleById(request.params.id);
+      if (!result) {
+        return response.status(404).json({
+          error: `Could not find raffle with id ${request.params.id}`,
+        });
+      } else if (result?.winner_id != null)
+        return response.status(400).json({
+          error: `raffle with id ${request.params.id} is not accepting new participants`,
+        });
+    } catch (error) {
+      return response.status(500).json({ error: error.message });
+    }
+
+    try {
+      await addNewParticipant(r);
+      const newRowJt = await addNewRaffleParticipant(request.params.id, r);
+      if (newRowJt?.jt_id) response.status(201).json({ data: newRowJt });
+      else
+        response.status(400).json({
+          error: `invalid participant for raffle id ${request.params.id}`,
+        });
```

Comment:
I think it's okay to pull these lines into the previous `try / catch` but let me know if I'm missing something.